wagtail/permission_policies/pages.py

```diff
@@ -0,0 +1,117 @@
+from django.contrib.auth import get_user_model
+from django.db.models import Q
+
+from wagtail.models import GroupPagePermission, Page
+from wagtail.permission_policies.base import BasePermissionPolicy
+
+
+class PagePermissionPolicy(BasePermissionPolicy):
+    perm_cache_name = "_page_perm_cache"
+
+    def __init__(self, model=Page):
+        super().__init__(model=model)
+
+    def get_all_permissions_for_user(self, user):
+        if not user.is_active or user.is_anonymous or user.is_superuser:
+            return GroupPagePermission.objects.none()
+        return GroupPagePermission.objects.filter(group__user=user).select_related(
+            "page"
+        )
+
+    def _base_user_has_permission(self, user):
+        if not user.is_active:
+            return False
+        if user.is_superuser:
+            return True
+        return None
+
+    def user_has_permission(self, user, action):
+        return self.user_has_any_permission(user, [action])
+
+    def user_has_any_permission(self, user, actions):
+        base_permission = self._base_user_has_permission(user)
+        if base_permission is not None:
+            return base_permission
+        permissions = {
+            perm.permission_type for perm in self.get_cached_permissions_for_user(user)
+        }
```

Comment:
This needs to incorporate the special case that an "add" permission should be sufficient for `user_has_any_permission(user, ["edit"])` to return true (since they can edit their own instances). This makes it consistent with `OwnershipPermissionPolicy`.

Reply:
Good catch, thanks! I missed this in 4e3c7d1. Fixed in 7265fee.